src/App.tsx

```diff
@@ -1,56 +1,65 @@
-import React from 'react';
+import React, { useState, useMemo, useCallback } from 'react';
+import cn from 'classnames';
+import debounce from 'lodash.debounce';
 import './App.scss';
+
 import { peopleFromServer } from './data/people';
+import { PeopleList } from './components/PeopleList';
+import { Person } from './types/Person';
+
+const DELAY = 300;
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [inputValue, setInputValue] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [isListVisible, setIsListVisible] = useState(false);
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, DELAY), []);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInputValue(event.target.value);
+    applyQuery(event.target.value);
+  };
+
+  const filteredPeople = useMemo(
+    () =>
+      // eslint-disable-next-line implicit-arrow-linebreak
+      peopleFromServer.filter((person) => person.name.toLowerCase()
+        .includes(appliedQuery
+          .toLowerCase())),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [...peopleFromServer, appliedQuery],
+  );
```

Comment:
No need to filter in case query is empty.
Avoid using `eslint-disable-next-line react-hooks/exhaustive-deps`